src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs

```diff
@@ -92,5 +93,29 @@ public static IWebHostBuilder UseDefaultServiceProvider(this IWebHostBuilder hos
                 services.Replace(ServiceDescriptor.Singleton<IServiceProviderFactory<IServiceCollection>>(new DefaultServiceProviderFactory(options)));
             });
         }
+
+        /// <summary>
+        /// Runs a web application with the specified handler
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to run.</param>
+        /// <param name="handler">A delegate that handles the request.</param>
+        public static IWebHost Run(this IWebHostBuilder hostBuilder, RequestDelegate handler)
+        {
+            var host = hostBuilder.Configure(app => app.Run(handler)).Build();
+            host.Start();
+            return host;
+        }
+
+        /// <summary>
+        /// Runs a web application with the specified handler
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to run.</param>
+        /// <param name="configure">The delegate that configures the <see cref="IApplicationBuilder"/>.</param>
+        public static IWebHost RunWith(this IWebHostBuilder hostBuilder, Action<IApplicationBuilder> configure)
```

Comment:
Once an IApplicationBuilder is needed it's probably alright to call Configure and Build separately. If another Run API then maybe one that takes Run(handler, port) to avoid the extra UseUrls. Also like the idea of having Kestrel as a default.